modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/GridDhtPartitionsStateValidator.java

```diff
@@ -96,13 +92,27 @@ public void validatePartitionCountersAndSizes(
                 ignoringNodes.add(id);
         }
 
+        StringBuilder error = new StringBuilder();
+
+        Map<Integer, Map<UUID, Long>> resSize = validatePartitionsSizes(top, messages, ignoringNodes);
+
+        AffinityTopologyVersion topVer = fut.context().events().topologyVersion();
+
+        if (!resUpdCnt.isEmpty() && !resSize.isEmpty()) {
+            error.append("Partitions cache size and update counters are inconsistent for ")
+                .append(fold(topVer, resUpdCnt, resSize));
+        }
+
         if (!cctx.cache().cacheGroup(top.groupId()).mvccEnabled()) { // TODO: Remove "if" clause in IGNITE-9451.
             // Validate cache sizes.
-            result = validatePartitionsSizes(top, messages, ignoringNodes);
-
-            if (!result.isEmpty())
-                throw new IgniteCheckedException("Partitions cache sizes are inconsistent for " + fold(topVer, result));
+            if (!resUpdCnt.isEmpty() && resSize.isEmpty())
```

Comment:
I think it will be correct to call validatePartitionsSizes here, and generate the message after.